### PR TITLE
feat: Make the Router lookup method available for others to use

### PR DESCRIPTION
**Pull request by Artur-**
When building a navigation menu component which is not based on RouterLink, there is the same need as RouterLink has: to create links to certain views. That requires a Router reference.

### CODE REVIEW

**Comment by mshabarov** [comment]
Could you please elaboration more why we need to expose the `Router`. I'm not against that, but just want to clarify why `UI::navigate` is not enough. Is there an issue/request for it?

**Comment by Artur-** [comment]
When building a navigation menu component which is not based on `RouterLink`, there is the same need as `RouterLink` has: to create links to certain views. That requires a `Router` reference

File: flow-server/src/main/java/com/vaadin/flow/component/ComponentUtil.java [hunk 1]
```diff
@@ -602,4 +604,31 @@ public static <T> T getData(Component component, Class<T> type) {
                 (attributes, ignore) -> attributes.getAttribute(type), type);
     }
 
+    /**
+     * Gets the router instance for the given component.
+     *
+     * Falls back to the router for the currently active VaadinService if the
+     * component is not attached.
+     *
+     * @return a router instance
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/component/ComponentUtil.java:613]:
```suggestion
     * @param component component for which the requested router instance serves navigation
     * @return a router instance
```

**Review by mshabarov — APPROVED** [review]

**Comment by mshabarov** [comment]
> When building a navigation menu component which is not based on RouterLink, there is the same need as RouterLink has: to create links to certain views. That requires a Router reference

Thanks! I've updated the description.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**